e2e/aws/eks_test.go

```diff
@@ -90,7 +91,14 @@ func awsEKSDiscoveryMatchedCluster(t *testing.T) {
 		defer cancel()
 
 		clusters, err := authC.GetKubernetesClusters(ctx)
-		return err == nil && len(clusters) == 1 && clusters[0].GetName() == os.Getenv(discoveredClusterNameEnv)
+		if err != nil {
+			return false
+		}
+		// Fail fast if the discovery service creates more than one cluster.
+		assert.Equal(t, 1, len(clusters))
+		// Fail fast if the discovery service creates a cluster with a different name.
+		assert.Equal(t, os.Getenv(discoveredClusterNameEnv), clusters[0].GetName())
+		return true
```

Comment:
how does this work with the eventually func? The assert.Equal calls just return a boolean and log the failure, so the function only runs once?

Reply:
I forgot to force push lol